tools/integration_tests/managed_folders/view_permissions_test.go

```diff
@@ -0,0 +1,218 @@
+			if len(objs) != NumberOfObjectsInDirForListTest {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), NumberOfObjectsInDirForListTest, len(objs))
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder1  -- ManagedFolder1
+			if objs[0].Name() != ManagedFolder1 || objs[0].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", EmptyManagedFolder1, objs[0].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder2     -- ManagedFolder2
+			if objs[1].Name() != ManagedFolder2 || objs[1].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", EmptyManagedFolder2, objs[1].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/simulatedFolder   -- SimulatedFolder
+			if objs[2].Name() != SimulatedFolder || objs[2].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", SimulatedFolder, objs[2].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/testFile  -- File
+			if objs[3].Name() != File || objs[3].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+			return nil
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == ManagedFolder1 {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder1/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == ManagedFolder2 {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder2/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == SimulatedFolder {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/simulatedFolder/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		t.Errorf("error walking the path : %v\n", err)
+		return
+	}
+
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
```

Comment:
Can we write the comment, what scenario we are trying to test here?

What I understood: View permission on bucket and [nil or view] permission on managed folder?